R/zchunk_LA111.rsrc_fos_Prod.R

```diff
@@ -1,89 +1,223 @@
+
+    # Determine regional shares of production for each primary fuel
+    L1011.en_bal_EJ_R_Si_Fi_Yh %>%
+      filter(sector == "out_resources", fuel %in% energy.RSRC_FUELS, year %in% HISTORICAL_YEARS) ->
+      L111.Prod_EJ_R_F_Yh_IEA
+
+    L111.Prod_EJ_R_F_Yh_IEA %>%
+      group_by(sector, fuel, year) %>%
+      mutate(share = value / sum(value)) %>%
+      select(-value) ->
+      L111.Prod_share_R_F_Yh
+
+    # Multiply through to calculate production by fuel
+    L111.Prod_EJ_R_F_Yh_IEA %>%
+      select(-value) %>%
+      left_join_error_no_match(select(L111.TPES_EJ_F_Yh, fuel, year, value), by = c("fuel", "year")) %>%
+      left_join_error_no_match(L111.Prod_share_R_F_Yh, by = c("GCAM_region_ID", "sector", "fuel", "year")) %>%
+      mutate(value = value * share) %>%
+      select(-share) ->
+      L111.Prod_EJ_R_F_Yh_IEA_adj
+
+    # Determine unconventional oil production (58-67)
+    rsrc_unconv_oil_prod_bbld %>%
+      gather(year, value, matches(YEAR_PATTERN)) %>%
+      mutate(year = as.integer(year)) %>%
+      # interpolate production to all historical years
+      complete(iso, year = HISTORICAL_YEARS) %>%
+      arrange(iso, year) %>%
+      group_by(iso) %>%
+      mutate(value = approx_fun(year, value, rule = 2)) %>%
+      left_join_error_no_match(select(iso_GCAM_regID, iso, GCAM_region_ID), by = "iso") %>%
+      # summarise by GCAM region and convert to EJ/yr
+      group_by(GCAM_region_ID, year) %>%
+      summarise(value = sum(value) * CONV_BBLD_EJYR) %>%
+      mutate(sector = "unconventional oil production",
+             fuel = "unconventional oil") ->
+      L111.Prod_EJ_ctry_unconvOil_Yh
+
+    # Subtract the unconventional oil, append the unconventional oil to the table, and write it out
+    # Deduct unconventional oil from total oil and include unconventional oil in calibrated production table
+    # Start by making an 'unconventionals' flag to easily split data
+    L111.Prod_EJ_R_F_Yh_IEA_adj %>%
+      mutate(unconventionals = GCAM_region_ID %in% L111.Prod_EJ_ctry_unconvOil_Yh$GCAM_region_ID &
+               fuel == "refined liquids") ->
+      L111.Prod_EJ_R_F_Yh
+
+    L111.Prod_EJ_R_F_Yh %>%
+      filter(unconventionals) %>%
+      left_join_error_no_match(select(L111.Prod_EJ_ctry_unconvOil_Yh, GCAM_region_ID, year, value),
+                               by = c("GCAM_region_ID", "year")) %>%
+      mutate(value = value.x - value.y) %>%
+      bind_rows(filter(L111.Prod_EJ_R_F_Yh, !unconventionals)) %>%
+      select(-value.x, -value.y, -unconventionals) %>%
+      # switch the names from final to primary
+      mutate(fuel = if_else(fuel == "refined liquids", "crude oil", fuel),
+             fuel = if_else(fuel == "gas", "natural gas", fuel)) %>%
+      bind_rows(select(L111.Prod_EJ_ctry_unconvOil_Yh, GCAM_region_ID, sector, fuel, year, value)) ->
+      L111.Prod_EJ_R_F_Yh
+
+    # ------- FOSSIL RESOURCE SUPPLY CURVES
+
+    # Using supply curves from GCAM 3.0 (same as MiniCAM) (83-93)
```

Comment:
See line 51 above: they're references to line numbers in the original data system file. I got the idea from @abigailsnyder .